### ISSUE TITLE
Highlight states with warnings/errors in the state diagram

### ISSUE THREAD

**Issue by Meorge** [AutomatonBuilderGUI/AutomatonBuilderGUI#86]
Currently, when a state has an error, it only shows up in the sidebar, like so:

![CleanShot 2024-08-20 at 15 11 42@2x](https://github.com/user-attachments/assets/b78279da-d3f5-41b7-8622-d0d9d5443714)

Since the user is mostly interacting with the state nodes directly in the main area of the screen, it would be nice to reflect the nodes with errors there too. Here's a mockup of one possible approach:

<img width="1072" alt="State Error Mockup" src="https://github.com/user-attachments/assets/0c7f546f-5ee8-476a-bf37-9d3e82bde95a">

Because state q3 has an error, it is given a red outline and tint, and error icon. The user can then identify it as having a problem at a quick glance.

Note that the specific way I drew the state with the error is just one idea. We could choose any number of approaches:
- Red tint only
- Red outline only
- Red tint and outline, but no error icon
- Error icon only
- etc

The main idea here is just to have some kind of indication on the state diagram, that a given state has an issue that needs to be fixed.

**Comment by Fenic27** [comment]
Hello :) I would like to be assigned to this issue.

**Comment by Meorge** [comment]
@Fenic27 Thanks! I will assign you to this issue.

Given that it has a larger design aspect to it (figuring out how exactly the errored states and transitions should look), this may be something where we discuss prototypes and make modifications before a final merge is performed. As you make progress, please feel free to post screenshots of your current designs so that others can give feedback and we can arrive at a design we think looks good. 😄

**Comment by Fenic27** [comment]
I did few designs. Let me know your opinion which one do you like the most. :) 
![Uploading Screenshot 2024-11-05 at 13.50.13.png…]()

**Comment by Fenic27** [comment]
I did few designs. Let me know your opinion which one do you like the most. :)

<img width="563" alt="Screenshot 2024-11-05 at 13 58 29" src="https://github.com/user-attachments/assets/7edf198f-fe97-4e77-800c-acb2d9b85669">
<img width="673" alt="Screenshot 2024-11-05 at 13 50 13" src="https://github.com/user-attachments/assets/db52c6fc-85ca-403a-9ce1-4a936ca342e9">
<img width="626" alt="Screenshot 2024-11-05 at 13 53 28" src="https://github.com/user-attachments/assets/d35f611c-e2c6-4d14-a7af-70c071fc6b42">
<img width="428" alt="Screenshot 2024-10-29 at 15 19 33" src="https://github.com/user-attachments/assets/1ae6aec1-5f19-4abe-a2a8-df34a8e1d730">
<img width="564" alt="Screenshot 2024-10-29 at 16 23 46" src="https://github.com/user-attachments/assets/d65600fc-096a-4bf0-99d2-09cfb4f4590c">

**Comment by Meorge** [comment]
Thanks for putting those together!

Out of all of these, I prefer the ones in the first screenshot the most. The colors look a bit easier on the eyes to me than some of the other ones, which are stronger. I think it'd make sense to synchronize the colors used there with those used in the error boxes that appear on the left-hand panel for visual consistency.
Having the X icon is good for people with colorblindness and other visual disabilities, but I think its current position might take up a bit more space below the node than is desirable. How does it look when placed on the edge of the node instead?

I think it'll be good to get some others' thoughts on the designs before deciding to move forward with one version as well :)

**Comment by noahwagner04** [comment]
@Fenic27 I personally like your first design the most, and Malcolm's design as well. I agree with with Malcolm though, I think having the "X" be nested on the state might look better.

**Comment by Fenic27** [comment]
<img width="415" alt="final" src="https://github.com/user-attachments/assets/12e3a0fb-2ff9-401c-bf52-a52d1c907f50">
Ok, I fixed the colours and move the circle with "X".

**Comment by noahwagner04** [comment]
Yeah that looks really good. I think the "X" would look better if they centered on the bottom edge of the nodes (or just a little bit higher). Thats just my opinion though, looks good overall.

**Comment by Fenic27** [comment]
I agree,  I think this one is the best one :)

<img width="525" alt="final" src="https://github.com/user-attachments/assets/fa078902-a202-4b2c-8651-aeb19a86ba1d">

**Comment by micahwagner** [comment]
Yeah thats definitely the best one

**Comment by Meorge** [comment]
I agree, I think that most recent revision looks great! 😄

**Comment by Meorge** [comment]
@Fenic27 I believe those files are part of `automaton-ts`, the repo for which is here: https://github.com/Meorge/automaton-ts You can fork it there, make the changes, and submit a PR that I can merge into that repo. Given how closely connected that library is to the GUI project, I should transfer it to this organization at some point, but for now it is there on my GitHub profile.
